### ISSUE TITLE
Enhancement: support for file:// urls for local filesystem objects

### ISSUE THREAD

**Issue by aplsysmngr** [bartreardon/Dialog#58]
It would be nice to open a local object in filesystem, like an app or other executable from a button or a link.
Like opening Munki MSC for updates:
file:///Applications/Managed Software Center.app

**Comment by bartreardon** [comment]
This is the function of the various return codes though as detailed in the [wiki](https://github.com/bartreardon/Dialog-public/wiki/Capturing-Dialog-output-for-further-processing#return-codes). So if you wanted to open MSC you could tie the exit code to the button of your choice:

```bash
dialog --button2text "Open MSC"
case $? in
  0)
  echo "Pressed OK"
  ;;
  2)
  echo "Opening MSC"
  open -a "/Applications/Managed Software Center.app"
  ;;
  *)
  echo "Something else happened"
  ;;
esac
```

There is the special "experimental" `--button1shellaction` command which will perform the specified shell command upon selection of the default button.

**Comment by bartreardon** [comment]
Just adding, in addition to processing the exit code, you can include links into the markdown to open MSC or Self Service or a `file://` 

e.g. `dialog --title "Links" --message "[Open MSC](munki://category-all)\n\n[Open Self Service](jamfselfservice://content?entity=policy&id=1&action=view)" `

<img width="727" alt="image" src="https://user-images.githubusercontent.com/3598965/139532641-98ec0e74-3b2c-47f6-a25a-a76423d8c7e7.png">

**Comment by aplsysmngr** [comment]
Hmm. On my Mac this gets always 0 as return code so Cancel button is not handled at all:
`dialog --button1text "Open MSC" --button2text "Cancel"
case $? in
  0)
  	echo "Opening MSC"
  	open -a "/Applications/Managed Software Center.app"
  	;;
  2)
  	echo "Pressed Cancel"
  	;;
  *)
  	echo "Something else happened"
  	;;
esac`

**Comment by bartreardon** [comment]
doesn't sound right. Whats your actual code look like? You should be capturing `$?` immediately after running dialog. if you want to do _anything else_ prior to using the exit code capture it with something like `EXITCODE=$?` and then perform the case statement on `${EXITCODE}`. But of you do `dialog ... ; something` or `dialog ... | something` then you won't be capturing the exit status of dialog

**Comment by bartreardon** [comment]
@aplsysmngr there was a big in 1.8.0 - can you verify this build works as you'd expect? [dialog-1.8.1.zip](https://github.com/bartreardon/Dialog-public/files/7455904/dialog-1.8.1.zip)

**Comment by bartreardon** [comment]
fixed in the 1.8.1 release https://github.com/bartreardon/Dialog/releases/tag/v1.8.1
